src/PrestaShopBundle/Resources/config/services/adapter/currency.yml

```diff
@@ -17,3 +17,44 @@ services:
             - '@=service("prestashop.adapter.legacy.configuration").get("PS_CURRENCY_DEFAULT")'
         tags:
             - { name: 'tactician.handler', command: 'PrestaShop\PrestaShop\Core\Domain\Currency\Command\DeleteCurrencyCommand' }
+
+    prestashop.adapter.currency.command_handler.create_currency:
+        class: 'PrestaShop\PrestaShop\Adapter\Currency\CommandHandler\AddCurrencyHandler'
+        tags:
+            - { name: 'tactician.handler', command: 'PrestaShop\PrestaShop\Core\Domain\Currency\Command\AddCurrencyCommand' }
+
+    prestashop.adapter.currency.command_handler.update_currency:
+        class: 'PrestaShop\PrestaShop\Adapter\Currency\CommandHandler\EditCurrencyHandler'
+        arguments:
+          - '@=service("prestashop.adapter.legacy.configuration").get("PS_CURRENCY_DEFAULT")'
+          - '@=service("prestashop.adapter.multistore_feature").isUsed()'
+        tags:
+            - { name: 'tactician.handler', command: 'PrestaShop\PrestaShop\Core\Domain\Currency\Command\EditCurrencyCommand' }
+
+    prestashop.adapter.currency.command_handler.update_exchange_rates:
+        class: 'PrestaShop\PrestaShop\Adapter\Currency\CommandHandler\RefreshExchangeRatesHandler'
+        tags:
+            - { name: 'tactician.handler', command: 'PrestaShop\PrestaShop\Core\Domain\Currency\Command\RefreshExchangeRatesCommand' }
+
+    prestashop.adapter.currency.command_handler.update_live_exchange_rates:
+        class: 'PrestaShop\PrestaShop\Adapter\Currency\CommandHandler\ScheduleExchangeRatesUpdateHandler'
+        arguments:
+          - '@prestashop.adapter.legacy.configuration'
+          - '@=service("prestashop.adapter.legacy.context").getContext().shop'
+          - '@prestashop.adapter.shop.shop_url'
+          - '@translator'
+          - '@=service("prestashop.adapter.data_provider.module").isInstalled("cronjobs")'
```

Comment:
I'm worried because we have hardcoded a module name here in the SF config.

There is definitely something wrong with this module that is finally very coupled with the Core 😅 but I would like to avoid hardcoded values into the Symfony configuration if possible. I'd prefer to pass the `prestashop.adapter.data_provider.module` service and have the hardcoded module name in the code.

Reply:
I'm using `@=service("prestashop.adapter.data_provider.module").isInstalled("cronjobs")` in two places - Core class `CurrencyExchangeRateType` and in Adapter `ScheduleExchangeRatesUpdateHandler`.

Unfortunately this class does not have any interface and it is used in Adapter. An Adapter usage in Core should be avoided at all costs as I learned :thinking: 

When we get rid of the module usage instead of passing `@=service("prestashop.adapter.data_provider.module").isInstalled("cronjobs")`  we can pass some kind of configuration which defines if this feature is on or off or we will remove it anyway and it will be just one line to remove when doing ctrl +F across whole project. If I use it differently in Adapter there is a greater change that somewhere this might be forgotten :smiley:

Reply:
That makes sense 👍 let's leave it like this